resources/tests/test_reservation_api.py

```diff
@@ -898,60 +897,70 @@ def test_state_filters(user_api_client, user, list_url, reservations_in_all_stat
 
 @override_settings(RESPA_MAILS_ENABLED=True)
 @pytest.mark.django_db
-def test_requested_mail(staff_api_client, staff_user, list_url, resource_in_unit, reservation,
+def test_requested_mail(staff_api_client, staff_user, user_api_client, list_url,
                         reservation_data_extra):
-    resource_in_unit.need_manual_confirmation = True
-    resource_in_unit.save()
-    assign_perm('can_approve_reservation', staff_user, reservation.resource.unit)
+    resource = Resource.objects.get(id=reservation_data_extra['resource'])
+    resource.need_manual_confirmation = True
+    resource.save()
+    assign_perm('can_approve_reservation', staff_user, resource.unit)
     reservation_data_extra['state'] = Reservation.REQUESTED
 
-    response = staff_api_client.post(list_url, data=reservation_data_extra, format='json', HTTP_ACCEPT_LANGUAGE='en')
+    response = user_api_client.post(list_url, data=reservation_data_extra, format='json')
     assert response.status_code == 201
     assert len(mail.outbox) == 1
     mail_instance = mail.outbox[0]
     assert 'Reservation requested' in mail_instance.subject
     assert len(mail_instance.to) == 1
-    assert mail_instance.to[0] == 'test.reserver@test.com'
+    assert mail_instance.to[0] == reservation_data_extra['reserver_email_address']
     mail_message = str(mail_instance.message())
-    assert 'You have created a preliminary reservation' in mail_message
+    assert 'made a preliminary reservation' in mail_message
```

Comment:
I think these mail related asserts could be done using the helper `check_received_mail()`